include/crocoddyl/multibody/costs/contact-cop-position.hxx

```diff
@@ -0,0 +1,92 @@
+                                          boost::shared_ptr<ActivationModelAbstract> activation,
+                                          const std::size_t& frame_id, const Vector2s& cop_region, 
+                                          const Vector3s& normal, const std::size_t& nu)
+    : Base(state, activation, nu), frame_id_(frame_id), cop_region_(cop_region), normal_(normal) {
+    }
+
+template <typename Scalar>
+CostModelContactCoPPositionTpl<Scalar>::~CostModelContactCoPPositionTpl() {}
+
+template <typename Scalar>
+void CostModelContactCoPPositionTpl<Scalar>::calc(const boost::shared_ptr<CostDataAbstract>& data,
+                                           const Eigen::Ref<const VectorXs>&, const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+
+  // Transform the contact force
+  d->f = d->contact->jMf.actInv(d->contact->f);
+
+  // Compute the cost residual respecting A * f
+  data->r.noalias() = d->A * d->f.toVector();
+
+  // Compute the cost
+  activation_->calc(data->activation, data->r);
+  data->cost = data->activation->a_value;
+}
+
+template <typename Scalar>
+void CostModelContactCoPPositionTpl<Scalar>::calcDiff(const boost::shared_ptr<CostDataAbstract>& data,
+                                                      const Eigen::Ref<const VectorXs>&, 
+                                                      const Eigen::Ref<const VectorXs>&) {
+  // Update all data
+  Data* d = static_cast<Data*>(data.get());
+
+  // Get the derivatives of the contact wrench
+  const MatrixXs& df_dx = d->contact->df_dx;
+  const MatrixXs& df_du = d->contact->df_du;
+  const Matrix46s& A = d->A;
+
+  //Compute the derivatives of the activation function
+  activation_->calcDiff(data->activation, data->r);
+
+  //Compute the derivatives of the cost residual
+  data->Rx.noalias() = A * df_dx;
+  data->Ru.noalias() = A * df_du;
+  d->Arr_Ru.noalias() = data->activation->Arr * data->Ru;
+
+  //Compute the first order derivatives of the cost function
+  data->Lx.noalias() = data->Rx.transpose() * data->activation->Ar;
+  data->Lu.noalias() = data->Ru.transpose() * data->activation->Ar;
+
+  //Compute the second order derivatives of the cost function
+  data->Lxx.noalias() = data->Rx.transpose() * data->activation->Arr * data->Rx;
+  data->Lxu.noalias() = data->Rx.transpose() * d->Arr_Ru;
+  data->Luu.noalias() = data->Ru.transpose() * d->Arr_Ru;
+}
+
+template <typename Scalar>
+boost::shared_ptr<CostDataAbstractTpl<Scalar> > CostModelContactCoPPositionTpl<Scalar>::createData(
+    DataCollectorAbstract* const data) {
+  return boost::allocate_shared<Data>(Eigen::aligned_allocator<Data>(), this, data);
+}
+
+template<typename Scalar>
+const std::size_t& CostModelContactCoPPositionTpl<Scalar>::get_frame_id() const {
+  return frame_id_;
+}
+
+template<typename Scalar>
+const typename MathBaseTpl<Scalar>::Vector2s& CostModelContactCoPPositionTpl<Scalar>::get_cop_region() const {
```

Comment:
change the name here to support_area as well

Reply:
See above comment